custom/icds/management/commands/copy_icds_app.py

```diff
@@ -23,3 +23,18 @@ def handle(self, domain, app_id, version, new_name, **options):
         old_app = wrap_app(old_app)
         old_app.convert_build_to_app()
         new_app = import_app(old_app.to_json(), domain, source_properties={'name': new_name})
+
+        old_to_new = get_old_to_new_config_ids(old_app, new_app)
+        for form in new_app.get_forms():
+            for old_id, new_id in old_to_new:
+                form.source = form.source.replace(old_id, new_id)
+
+        new_app.save()
```

Comment:
For some reason I thought you had to call `save_xform` directly to update a form. You tested this command out, right?

Reply:
Yeah, it worked fine, though now I'm gonna test it again to be doubly sure.

Reply:
Here's one of the things I referenced in writing this:
https://github.com/dimagi/commcare-hq/blob/45cab596a274f242d66b6372a7678fc5293853de/corehq/apps/app_manager/management/commands/convert_basic_module_to_advanced.py#L147-L166
it similarly modifies `form.source` then saves the app.

Looking at `save_xform`, it seems this is the operative line:
https://github.com/dimagi/commcare-hq/blob/45cab596a274f242d66b6372a7678fc5293853de/corehq/apps/app_manager/util.py#L181-L181
Though despite the name, I don't think it actually saves the form - I looked at a couple places its used and they later call `app.save()`.  Is this meant to overwrite the app xml then?

Reply:
Now that I'm actually looking at the code instead of leaving vague review comments, yeah, I agree with you, the unicode handling is the only part that looks relevant. My memory was backwards, apparently the counterintuitive thing about `save_xform` is that it *doesn't* save.